lib/neo4j/shared/enum.rb

```diff
@@ -3,6 +3,7 @@ module Enum
     extend ActiveSupport::Concern
 
     class ConflictingEnumMethodError < Neo4j::Error; end
+    class InvalidEnumValueError < Neo4j::InvalidParameterError; end
```

Comment:
I'm inheriting from `InvalidParameterError` rather than `ArgumentError`. When I suggested `ArgumentError`, I didn't know about `InvalidParameterError`. Seeing as we're talking about passing an invalid enum value through a param, this seemed more appropriate. Let me know if you prefer `ArgumentError` though.